plugins/package-managers/bazel/src/main/kotlin/Bazel.kt

```diff
@@ -163,11 +204,37 @@ class Bazel(
         return null
     }
 
-    private fun getPackages(scopes: Set<Scope>, registry: BazelModuleRegistryService): Set<Package> {
+    /**
+     * Get the packages for the dependencies of the given [scopes] using the given [registry], for the dependencies that
+     * do not have a local path override. The dependencies having an override defined in [localPathOverrides] are
+     * created without querying [registry] and will use [projectVcs] as VCS information.
+     */
+    private fun getPackages(
+        scopes: Set<Scope>,
+        registry: BazelModuleRegistryService,
+        localPathOverrides: Map<String, String>,
+        projectVcs: VcsInfo
+    ): Set<Package> {
         val ids = scopes.collectDependencies()
-        val moduleMetadataForId = ids.associateWith { getModuleMetadata(it, registry) }
-        val moduleSourceInfoForId = ids.associateWith { getModuleSourceInfo(it, registry) }
-        return ids.mapTo(mutableSetOf()) { getPackage(it, moduleMetadataForId[it], moduleSourceInfoForId[it]) }
+        val (packageIdsWithPathOverride, otherPackageIds) = ids.partition { it.name in localPathOverrides }
+
+        val packagesWithPathOverrides = packageIdsWithPathOverride.map {
+            Package(
+                id = it,
+                declaredLicenses = emptySet(),
+                description = "",
+                homepageUrl = "",
+                binaryArtifact = RemoteArtifact.EMPTY,
+                sourceArtifact = RemoteArtifact.EMPTY,
+                vcs = projectVcs.copy(path = localPathOverrides.getValue(it.name))
+            )
+        }
+
+        val moduleMetadataForId = otherPackageIds.associateWith { getModuleMetadata(it, registry) }
+        val moduleSourceInfoForId = otherPackageIds.associateWith { getModuleSourceInfo(it, registry) }
+        return otherPackageIds.mapTo(mutableSetOf()) {
+            getPackage(it, moduleMetadataForId[it], moduleSourceInfoForId[it])
+        } + packagesWithPathOverrides
```

Comment:
Only seeing this now... up to you, but in order to avoid this addition, we could make a mutable set and map to it directly here and above:

    val packagesWithPathOverrides = mutableSetOf<Package>()

    packageIdsWithPathOverride.mapTo(packagesWithPathOverrides) {
    // ...
    return otherPackageIds.mapTo(packageIdsWithPathOverride) {
    // ...

Reply:
I renamed the variable `result` for clarity.